packages/docusaurus-theme-classic/src/theme-classic.d.ts

```diff
@@ -211,7 +211,7 @@ declare module '@theme/CodeBlock/Line' {
 
   export interface Props {
     readonly line: Token[];
-    readonly highlight: boolean;
+    readonly classNames: string[] | undefined;
```

Comment:
mitigated feelings about this change, as it's not idiomatic React. Why is it better to inject `classNames: string[]` vs `className: string` ?

This can be interesting if you want to use `classNames.includes("x")`, but user can use `split(" ")` anyway

Reply:
Yes, I want to support metadata beyond actual class names, like `pillNumber` you mentioned. Passing in `classNames` should allow more flexibility. It's also more performant because (1) the class names don't get concatenated and then split again (2) all class names are concatenated in one pass instead of twice, once with all custom class names and once with the hardcoded ones

Reply:
ok 👍